### ISSUE TITLE
False positive about MagicNumber

### ISSUE THREAD

**Issue by Zustin** [checkstyle/checkstyle#12687]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#MagicNumber

```bash
/var/tmp $ javac YOUR_FILE.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
<module name="MagicNumber">
  <property name="tokens" value="NUM_DOUBLE, NUM_FLOAT"/>
  <property name="ignoreNumbers" value="0, 0.5, 1"/>
  <property name="ignoreFieldDeclaration" value="true"/>
  <property name="ignoreAnnotation" value="true"/>
</module>
  </module>
</module>

/var/tmp $ cat YOUR_FILE.java
class C1 {

  public void magicMethod() {
    double double_var1 = 0D;
    double_var1 = 1.0 + 2.0; // violation
    double double_magic1 = 1.5_0; // violation
    if (
      1.0 < 2.0
    ); // violation
    double_magic1 *= 1.5; // violation
  }
}

class C2 {

  Object anon = new Object() {
    public void magicMethod() {
      double double_var1 = 0D;
      double_var1 = 1.0 + 2.0;
      double double_magic1 = 1.5_0; // violation
      if (1.0 < 2.0);
      double_magic1 *= 1.5;
    }
  };
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.3.3-all.jar -c config.xml YOUR_FILE.java
Starting audit...
[ERROR] /xxx:5:25: '2.0' is a magic number. [MagicNumber]
[ERROR] /xxx:6:28: '1.5_0' is a magic number. [MagicNumber]
[ERROR] /xxx:8:13: '2.0' is a magic number. [MagicNumber]
[ERROR] /xxx:10:22: '1.5' is a magic number. [MagicNumber]
[ERROR] /xxx:20:30: '1.5_0' is a magic number. [MagicNumber]
Audit done.
Checkstyle ends with 5 errors.
```

**Describe what you expect in detail.**

The magicMethods in C1 and C2 are same. The analysis results should be consistent (both have 4 warnings). Hence, I think this is a false negative.

**Comment by rnveach** [comment]
When our code is [looking for the top variable definition](https://github.com/checkstyle/checkstyle/blob/54616f662130aa0ca5afe45f72d36cdefb78a781/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MagicNumberCheck.java#L526) it stops on the `Object anon` incorrectly thinking it found a field. `ignoreFieldDeclaration` says to ignore fields, so it doesn't report a violation.

**Comment by Zustin** [comment]
@rnveach Thanks for your reply. If so, the `anon` should not have any violation, but it actually has a warning, hence I think this is a false positive.

**Comment by romani** [comment]
Behavior should be same.

**Comment by Kevin222004** [comment]
I am on it

**Comment by rnveach** [comment]
@Kevin222004 Where are we on the other 5 PRs you have open? I excluded the 1 blocked PR, and don't know the status of the others.

I think at this point we should be focusing on finishing PRs, then opening more. Let me know if something is blocking you.

**Comment by Kevin222004** [comment]
@rnveach please look on gitter i message you

**Comment by rnveach** [comment]
Since you say one is pending from your side, I suggest you work on that. If you have been waiting more than a week on any PR, then please ping the reviewer. We will get to your PRs as time allows. I recommend to think twice about increasing the amount of work you take on as it will just get stuck in the review process, as if we don't have time for your current ones, then how will we have time for new ones.

**Comment by rnveach** [comment]
Fix was merged
